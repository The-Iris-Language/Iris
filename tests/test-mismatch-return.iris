class Bush () {
    public:
        int leaves;
        void print() {
            Olympus.println("bush initialized");
        }

        Animal sniff () {
            return new Dog();
        }
}

class Animal () {
    public:
        Bush shrub;
        int x;
        string noise(int volume) {
            return "ruh roh raggy";
        }

        Animal fun_a () {
            Animal a = new Animal();
            a.x = 4;
            return a;
       }

       int class_count(Animal a) {
            Olympus.println(a.noise(8));
            return 0;
       }

}

class Dog of Animal () {
    public:
        string noise(int volume) {
            return "ruff ruff";
        }
        Dog fun() {
            Dog d = new Dog();
            return d;
        }

        void fun_two(Animal a) {
            Olympus.println("yay");
        }
}

class Main () {
    public:
        int univ main() {
            Olympus.print("hello\n");
            int i = 12;
            Animal a = new Animal();
            
            a.x = 10;
            
            a = a.fun_a();
            
            
            if (a.x == 10) {
                Olympus.println("Got old instance");
            } else if (a.x == 4) {
                Olympus.println("Got new instance");
            }
           
            .. a = d.fun();
            .. a.x = 10;
            .. if (a.x == 10) {
            ..     Olympus.println(a.noise(1));
            .. }
            Dog d = new Dog();


            Bush b = new Bush();
            b.leaves = 10;
            Animal a = new Animal();
            Olympus.println("assign below");
            a.shrub = b;
            Bush c = a.shrub;
            if (c.leaves == 10) {
                Olympus.println("10 leaves");
            }

            a = b.sniff();
            Olympus.println(a.noise(8));

            string s = "HI";
            if (s == "HI") {
                Olympus.println("howdy");
            }


            int x = a.class_count(d);

            ..Animal another_animal;
            ..another_animal = new Animal();
            ..Olympus.println(another_animal.noise(3));
            ..Bush b = new Bush();
            ..a.shrub = b;
            ..Bush c = a.shrub;
            ..c.print();

            ..d.fun_two(d);
            return 0;
        }
}