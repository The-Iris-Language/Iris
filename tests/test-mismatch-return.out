hello\nGot new instance
assign below
10 leaves
ruff ruff
howdy
ruff ruff
